bgFixes - implement all feedback

McHardex opened the pull request:
#### What does this PR do?
- [x]  adjust UI designs
- [x]  adjust the upvote and downvote icon
- [x]  fix GitHub pages hosted link on GitHub repository
- [x]  make all routes similar
- [x]  change class name to sentence case
- [x]  Adjust test description
- [x]  handle API route error
- [x]  Add more assertions to test
- [x] isolate meetup validation to separate folder
- [x] change test description for the incorrect date
- [x] add more test assertions
- [x] make UI navbar fullscreen

#### How should this be manually tested?

checkout into this branch `feature/fix-feedback-errors`
start the application by running `npm start`

#### What are the relevant pivotal tracker stories?(if applicable)

[#163051637](https://www.pivotaltracker.com/story/show/163051637)

Commit:
bgFixes - implement all feedback

Adjust the pivotal tracker descriptions
Use pivotal epic to organize the stories
Adjust UI designs
Adjust the upvote and downvote icon
fix GitHub pages hosted link on GitHub repository
Make all routes similar
OOP should start with a capital letter
Adjust test description.
handle API route error
Add more assertion to test
Add pivotal story link to PR

[Finishes #163051637]

mikey2020 commented on controller/MeetupController.js:
```diff
@@ -0,0 +1,76 @@
+    if (!isValidDate(req.body.happeningOn)) return res.status(400).send({ message: 'the date must be in this format: mm-dd-yyy' });
+
+    if (!req.body.tags) return res.status(400).send({ message: 'tags is required' });
+    if (req.body.tags.length < 3) return res.status(400).send({ message: 'tags length must be greater than 3' });
```
Great job on your validations but don't you think the code would be cleaner if you separate all the validations to another file.

McHardex replied:
Cool. Got it. Will work on that

mikey2020 commented on startup/db/meetupDb.js:
```diff
@@ -1,18 +1,16 @@
     title: 'coders brings live',
     location: 'Abuja',
-    happeningOn: new Date(today.setHours(today.getHours() + 2000)),
+    happeningOn: '12-04-2016',
```
Why did you hard code this value?

McHardex replied:
Initially, I wasn't implementing the get upcoming meetup routes well, so I used that to set upcoming dates but, I have changed the setup to the one on line 6

mikey2020 commented on test/meetup.spec.js:
```diff
@@ -121,6 +133,80 @@ describe('Meetups List Api Exists', () => {
+        });
+    });
+
+    it('should fail on POST with date input', (done) => {
```
I think this should be incorrect date input

McHardex replied:
alright, I'm going to fix that

mikey2020 commented on test/question.spec.js:
```diff
@@ -5,13 +5,14 @@ import { expect } from 'chai';
         .get('/api/v1/questions')
         .end((err, res) => {
           expect(res.statusCode).to.equal(200);
+          expect(res.body).to.be.an('object');
```
Nice job here on trying to check what the response body is but you are just checking that it is an object what about checking specific values in the response body.

McHardex replied:
Alright, thank you. I will make the necessary changes

Commit:
isolate meetup validation to seperate folder
change test description for incorrect date
add more test assertions
make UI navbar fullscreen

	#

Commit:
bgFixes - implement all feedback on UI
redesign UI and fix responsiveness

mikey2020 commented on test/question.spec.js:
```diff
@@ -30,6 +32,8 @@ describe('Question Api Exists', () => {
         .end((err, res) => {
           expect(res.statusCode).to.equal(201);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('data');
```
Why are you only checking for the property data?

McHardex replied:
I've been having issues working my way around checking for the keys in my data object. I'm thinking the error is because I am using a dummy database because when I use `expect(res.body).to.have.keys({ 'title', 'body' })`, my test kept failing saying it's expecting an additional key '0'. So I decided to only check for the variable holding the object inside.

mikey2020 commented on test/question.spec.js:
```diff
@@ -65,6 +75,20 @@ describe('Question Api Exists', () => {
         .end((err, res) => {
           expect(res.statusCode).to.equal(200);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('data');
```
Same thing here, why are you only checking for the property data?

McHardex replied:
I've been having issues working my way around checking for the keys in my data object. I'm thinking the error is because I am using a dummy database because when I use `expect(res.body).to.have.keys({ 'title', 'body' })`, my test kept failing saying it's expecting an additional key '0'. So I decided to only check for the variable holding the object inside.

Commit:
bg-Fixes - implement unit test feedback
add more test assertions

Commit:
bgFixes - implement UI feedback
add logout button to necessary pages

mikey2020 commented on test/meetup.spec.js:
```diff
@@ -5,24 +5,15 @@ import { expect } from 'chai';
-        .end((err, res) => {
-          expect(res.statusCode).to.equal(200);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('data');
```
You didn't change it here @McHardex

mikey2020 replied:
Please take some time to review your code before pushing

McHardex replied:
Is it OK to just check if the keys inside data object exists?

Commit:
bgFixes - implement api feedback
add more test assertions
correct error handling messages
